### PR TITLE
Clarify how to deploy the cross-device infrastructure

### PR DESCRIPTION
**Pull request by laurentgrangeau**
The purpose of this PR is to clarify the instructions for deploying the cross-device part, addind some prefix to buckets generation and activate missing APIS

### CODE REVIEW

**Review by ferrarimarco — APPROVED** [review]

File: terraform/cross-device/storage.tf [hunk 1]
```diff
@@ -17,7 +17,7 @@ module "buckets" {
   version    = "5.0.0"
   project_id = data.google_project.project.project_id
   location   = var.region
-  prefix     = "fcp-${var.environment}"
+  prefix     = "fcp-${var.environment}-${random_id.bucket_prefix.hex}"
```
Comment (ferrarimarco) [inline terraform/cross-device/storage.tf:20]: You don't need to do this yourself. Check the cloud-storage module docs. It supports adding a randomized suffix.

https://registry.terraform.io/modules/terraform-google-modules/cloud-storage/google/latest

File: terraform/cross-device/storage.tf [hunk 2]
```diff
@@ -51,3 +51,7 @@ module "buckets" {
     }
   }]
 }
+
+resource "random_id" "bucket_prefix" {
```
Comment (ferrarimarco) [inline terraform/cross-device/storage.tf:55]: You don't need this, see my other comment above.

File: terraform/cross-device/version.tf [hunk 3]
```diff
@@ -19,5 +19,9 @@ terraform {
       source  = "hashicorp/google"
       version = ">= 4.69.1, < 6.0.0"
     }
+    random = {
```
Comment (ferrarimarco) [inline terraform/cross-device/version.tf:22]: You don't need this, see my other comment above.

**Review by ferrarimarco — CHANGES_REQUESTED** [review]

**Comment by ferrarimarco** [comment]
@laurentgrangeau ping :)

File: README.md [hunk 4]
```diff
@@ -152,6 +152,8 @@ the following examples in the GKE cluster:
 
 - [Distributed TensorFlow Federated training](./examples/federated-learning/tff/distributed-fl-simulation-k8s/README.md)
 
+You can also deploy a cross-device infrastructure by following [this README.md](./terraform/cross-device/README.md)
```
[the remark this answers is not in the document]
Comment (laurentgrangeau) [reply README.md]: Done

File: terraform/cross-device/README.md [hunk 5]
```diff
@@ -15,6 +15,10 @@ It creates:
 - Pubsub topics that act as buses for messages between microservices
 - Buckets for storing the trained models
 
+The following diagram shows the resources that you create and configure with the blueprint.
+
+![alt_text](../../assets/cross-device.svg "Resources created by the blueprint")
```
[the remark this answers is not in the document]
Comment (laurentgrangeau) [reply terraform/cross-device/README.md]: I tried to refactor the diagram to go through Private Acces but it became unreadable with all the flux between components

File: terraform/cross-device/README.md [hunk 6]
```diff
@@ -27,6 +31,8 @@ This example builds on top of the infrastructure that the
 [blueprint provides](../../README.md), and follows the best practices the
 blueprint establishes.
 
+First, you need to deploy the infrastructure as explained in the [main README](../../README.md#deploy-the-blueprint). It will deploy the base infrastructure. Then, you can deploy the cross-device blueprint.
```
[the remark this answers is not in the document]
Comment (laurentgrangeau) [reply terraform/cross-device/README.md]: Done

File: terraform/services.tf [hunk 7]
```diff
@@ -50,7 +50,9 @@ module "project-services" {
     "meshconfig.googleapis.com",
     "meshtelemetry.googleapis.com",
     "monitoring.googleapis.com",
+    "run.googleapis.com",
```
[the remark this answers is not in the document]
Comment (laurentgrangeau) [reply terraform/services.tf]: Done

File: terraform/cross-device/storage.tf (hunk 1 above)
Comment (laurentgrangeau) [reply terraform/cross-device/storage.tf]: Done

File: terraform/cross-device/storage.tf (hunk 2 above)
Comment (laurentgrangeau) [reply terraform/cross-device/storage.tf]: Done

File: terraform/cross-device/version.tf (hunk 3 above)
Comment (laurentgrangeau) [reply terraform/cross-device/version.tf:22]: Done

File: terraform/cross-device/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ferrarimarco) [reply terraform/cross-device/README.md]: Maybe you can do this:

1. Detach Artifact Registry from the Private Google Access box
2. Put the Artifact registry box inside the VPC, as you did with other services
3. Attach the Artifact Registry box to the Cloud Firewall Rules box
4. Rename Cloud Firewall Rules box to Cloud Firewall

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Hi @laurentgrangeau !

One small thing to check on the diagram, and we should be good to go!

Thanks!

**Review by ferrarimarco — APPROVED** [review]

**Pull request merged**